File: easy/2373_largest_local_values_in_a_matrix/solution.py
from typing import List


class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)

        maxLocal = [[0] * (n - 2) for _ in range(n - 2)]

        for i in range(n - 2):
            for j in range(n - 2):
                currMax = 0

                for u in range(i, i + 3):
                    for v in range(j, j + 3):
                        currMax = max(currMax, grid[u][v])

                maxLocal[i][j] = currMax

        return maxLocal

File: easy/2373_largest_local_values_in_a_matrix/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>>
  largestLocal(std::vector<std::vector<int>> &grid) {
    size_t n = grid.size();

    std::vector<std::vector<int>> maxLocal(n - 2, std::vector<int>(n - 2, 0));

    for (size_t i = 0; i < n - 2; ++i) {
      for (size_t j = 0; j < n - 2; ++j) {
        maxLocal[i][j] = std::max({
            grid[i][j],
            grid[i][j + 1],
            grid[i][j + 2],
            grid[i + 1][j],
            grid[i + 1][j + 1],
            grid[i + 1][j + 2],
            grid[i + 2][j],
            grid[i + 2][j + 1],
            grid[i + 2][j + 2],
        });
      }
    }

    return maxLocal;
  }
};
